velox/dwio/common/DirectBufferedInput.cpp

```diff
@@ -270,10 +270,18 @@ std::vector<cache::CachePin> DirectCoalescedLoad::loadData(bool isPrefetch) {
       buffers.push_back(folly::Range(request.tinyData.data(), region.length));
     }
     lastEnd = region.offset + request.loadSize;
-    size += std::min<int32_t>(loadQuantum_, region.length);
+    size += request.loadSize;
   }
-  input_->read(buffers, requests_[0].region.offset, LogType::FILE);
+
+  uint64_t usecs = 0;
+  {
+    MicrosecondTimer timer(&usecs);
+    input_->read(buffers, requests_[0].region.offset, LogType::FILE);
+  }
+
   ioStats_->read().increment(size);
+  ioStats_->incRawBytesRead(size - overread);
+  ioStats_->incTotalScanTime(usecs * 1'000);
```

Comment:
`queryThreadIoLatency` also needs to be increased

Reply:
@Yuhta Added. Can you help to review again?